internal/core/docker_context.go

```diff
@@ -0,0 +1,328 @@
+package core
```

Comment:
question: can we check this is all required, as it looks like some types don't seem to be referenced.

Reply:
The only exposed function in this file is `GetDockerHostFromCurrentContext`, which starts the call to the types and functions here.